### PR TITLE
59 ability to consume messages from superstream

### PR DESCRIPTION
**Pull request by tarzacodes**

### CODE REVIEW

**Review by icappello — APPROVED** [review]
lgtm

File: src/client.ts [hunk 1]
```diff
@@ -310,6 +321,11 @@ export class Client {
     return res.ok
   }
 
+  public async declareSuperStreamConsumer(superStream: string, handle: ConsumerFunc): Promise<SuperStreamConsumer> {
+    const partitions = await this.queryPartitions({ superStream })
+    return SuperStreamConsumer.create(handle, { locator: this, consumerRef: "test", partitions })
```
Comment (albertobarrila) [inline src/client.ts:326]: the consumerRef is hardcoded as "test", is it right?

File: src/publisher.ts [hunk 2]
```diff
@@ -82,6 +82,7 @@ export class StreamPublisher implements Publisher {
   private scheduled: NodeJS.Immediate | null
   private logger: Logger
   private maxChunkLength: number
+  private isClosed = false
```
Comment (albertobarrila) [inline src/publisher.ts:85]: I honestly do not like "is" properties, I would suggest using only closed, wdyt?

File: src/super_stream_consumer.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { Client } from "./client"
+import { Consumer, ConsumerFunc } from "./consumer"
+import { Offset } from "./requests/subscribe_request"
+
+export class SuperStreamConsumer {
+  private consumers: Map<string, Consumer> = new Map<string, Consumer>()
+  public consumerRef: string
+  private locator: Client
```
Comment (albertobarrila) [inline src/super_stream_consumer.ts:8]: is locator a good name for the client underneath the consumer?

Comment (tarzacodes) [reply src/super_stream_consumer.ts:8]: It is the locator, that's the term we are using for the client that manages the connections right?

**Pull request merged**